mm/slab.c

```diff
@@ -3529,7 +3529,10 @@ int kmem_cache_alloc_bulk(struct kmem_cache *s, gfp_t flags, size_t size,
 
 	local_irq_disable();
 	for (i = 0; i < size; i++) {
-		void *objp = __do_cache_alloc(s, flags);
+		void *objp = kfence_alloc(s, s->object_size, flags);
+
+		if (likely(!objp))
```

Comment:
In the kernel we may use the GNU C extension to the conditional operator (used all over the place): https://gcc.gnu.org/onlinedocs/gcc/Conditionals.html

So this could be
```
void *objp = kfence_alloc(s, s->object_size, flags) ?: __do_cache_alloc(s, flags);
```

I leave it to you if that's more readable or not.

Reply:
Hm, that's handy and probably more readable.